### DISCUSSION TITLE
Figma -> React Native w/ Unistyles

### DISCUSSION THREAD

**Discussion by TheUltDev** [jpudysz/react-native-unistyles#44] [Show and tell]
I've added Unistyles as the default styling library used in my Figma -> React Native plugin for those interested.

 I know you've seen this @jpudysz, but I wanted to share with others who find the project and also have somewhere to communicate on the progress on my Unistyles support.

Note: unistyles support is fully done but is not released yet as I'm doing a major refactor and adding a lot of new features. I will update this discussion when the release drops. Most likely Unistyles 2.0 will drop before and I will update to that before releasing.

Plugin:  https://figma-to-react-native.com
Source: https://github.com/kat-tax/figma-to-react-native

<img width="1381" alt="Screenshot 2023-10-20 at 6 26 43 PM" src="https://github.com/jpudysz/react-native-unistyles/assets/2079305/3f4a0c7d-8823-4f9a-964e-ff8809588779">

**Comment by TheUltDev** [comment]
@jpudysz the coolest part about this is I'm developing a UI kit in Figma that works hand in hand with this plugin.
That means Unistyles will have a UI Kit both in Figma and React Native [+web] when I'm done!

Fun fact, it's the reversed approach of Gluestack, which generates a Figma file from their RN components.

**Reply by jpudysz** [reply]
Awesome work!
I'd be happy to include you in the documentation once your plugin reaches beta or is close to release.

P.S.
In v.2.0, I plan to retain the same API as in v.1.0. While there will be a different method to register themes and breakpoints, the behavior of `useStyles` and `createStyleSheet` should remain consistent.

**Reply by TheUltDev** [reply]
@jpudysz I've release Unistyles support in v43: https://github.com/kat-tax/figma-to-react-native/pull/16

This can be considered beta, there are a few known issues, but they will be resolved shortly.

**Reply by jpudysz** [reply]
Just wondering why are you still using beta.6? 😅

**Reply by TheUltDev** [reply]
@jpudysz that's just a dev dependency for types, forgot to update it.

**Reply by jpudysz** [reply]
How it's going?

**Reply by TheUltDev** [reply]
@jpudysz working on a large update for the plugin. Mainly going from a single component approach to a project view with background compilation. The work contains better unistyles integration as well, but I have a couple more weeks left in development.

**Reply by TheUltDev** [reply]
Big release in a week. This update also completes unistyles support.
Here's a preview of the theme switching in action (no recompile needed):

https://github.com/jpudysz/react-native-unistyles/assets/2079305/8b1cc059-215f-4de8-bd7a-c4e3ca6329eb

**Reply by rikur** [reply]
@TheUltDev 2.0 is basically here. Eager to try this combo to rebuild my app components.

**Reply by TheUltDev** [reply]
@rikur I have a big release that will go out in the next couple days that will support 2.0

**Reply by jpudysz** [reply]
@TheUltDev I'm happy to host you here: https://reactnativeunistyles.vercel.app/show-case/projects/

**Comment by TheUltDev** [comment]
Released `v45` that includes 1.2.0 support

Working on 2.0.0 support now. (but output will be essentially the same)

https://figma-to-react-native.com
